amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/StreamConfig.java

```diff
@@ -15,14 +15,15 @@
 
 package software.amazon.kinesis.common;
 
-import lombok.Value;
+import lombok.Data;
 import lombok.experimental.Accessors;
 
-@Value
+@Data
 @Accessors(fluent = true)
 public class StreamConfig {
-    StreamIdentifier streamIdentifier;
-    InitialPositionInStreamExtended initialPositionInStreamExtended;
+    private final StreamIdentifier streamIdentifier;
+    private final InitialPositionInStreamExtended initialPositionInStreamExtended;
+    private String consumerArn;
```

Comment:
Why do we need to make consumerArn (and this class) mutable? It doesn't look like we're setting consumerArn anywhere

Reply:
This is made as mutable to provide chain setter for consumerArn, a standard setter pattern we use across KCL. This will be set by customer and consumed in FanoutRecordPublisher